src/muxed_account.js

```diff
@@ -0,0 +1,132 @@
+import isString from 'lodash/isString';
+
+import xdr from './generated/stellar-xdr_generated';
+import { Account } from './account';
+import { StrKey } from './strkey';
+import {
+  decodeAddressToMuxedAccount,
+  encodeMuxedAccountToAddress
+} from './util/decode_encode_muxed_account';
+
+/**
+ * Represents a muxed account for transactions and operations.
+ *
+ * A muxed (or *multiplexed*) account (defined rigorously in
+ * [CAP-27](https://stellar.org/protocol/cap-27) and briefly in
+ * [SEP-23](https://stellar.org/protocol/sep-23)) is one that resolves a single
+ * Stellar `G...`` account to many different underlying IDs.
+ *
+ * For example, you may have a single Stellar address for accounting purposes:
+ *   GA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJVSGZ
+ *
+ * Yet would like to use it for 4 different family members:
+ *   1: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAAGZFQ
+ *   2: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAALIWQ
+ *   3: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAAPYHQ
+ *   4: MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAAQLQQ
+ *
+ * This object makes it easy to create muxed accounts from regular accounts,
+ * duplicate them, retrieve the underlying IDs, etc. without mucking around with
+ * the raw XDR.
+ *
+ * Because muxed accounts are purely an off-chain convention, they all share the
+ * sequence number tied to the underlying G... account. Thus, this object
+ * *requires* an {@link Account} instance to be passed in, so that muxed
+ * instances of an account can collectively modify the sequence number whenever
+ * a muxed account is used as the source of a @{link Transaction} with {@link
+ * TransactionBuilder}.
+ *
+ * @constructor
+ *
+ * @param {Account}   account - the @{link Account} instance representing the
+ *     underlying G... address
+ * @param {string}    id      - a stringified uint64 value that represents the
+ *     ID of the muxed account
+ */
+export class MuxedAccount {
+  constructor(baseAccount, id) {
```

Comment:
Nit: The interface here seems a bit surprising (to me). I guess the interface expected is something like:
```js
// base account setup (as before)
const gAddr = "GB3KJPLFUYN5VL6R3GU3EGCGVCKFDSD7BEDX42HWG5BWFKB3KQGJJRMA";
const seq = 5;
const base = new Account(gAddr, seq);

// Create muxed accounts with pointer back to parent to track seq
const sub1 = base.createSubaccount(1);
const sub2 = new MuxedAccount(base, 2);

// Creating a sibling of a muxed account
const sub3 = sub2.baseAccount().createSubaccount(3);

// Parsing helper
const mAddr = "MA7QYNF7SOWQ3GLR2BGMZEHXAVIRZA4KVWLTJJFC7MGXUA74P7UJUAAAAAAAAAAAALIWQ";
const parsed = MuxedAccount.fromAddress(mAddr, seq); // or something

// Fails
new Account(mAddr, seq);
new MuxedAccount(gAddr, seq);
```

Reply:
Good suggestion re: `baseAccount().createSubAccount(id)`, but I'm not sure how feasible the `fromAddress` API is. The base account gets disconnected, so there's no way to associate it. Yeah we could have a warning of some sort, or do a fetch via `server.loadAccount()`, but those both feel hacky to me. 

However, there is clear value in being able to create an object from the M-address. The best approach in this case may be to throw on any sequence number-related methods if `this.account` isn't set, and introduce a `setBaseAccount()` sorta thing. Then, you'd need to do something like:

```js
const parsed = MuxedAccount.fromAddress(mAddr);
expect(() => parsed.incrementSequenceNumber()).to.throw();
const baseAccount = new Account(
    parsed.underlyingGAddress(), /* or equivalent, naming is hard */
    sequence);
parsed.setBaseAccount(baseAccount);
expect(() => parsed.incrementSequenceNumber()).not.to.throw();
```

Reply:
Ohh wait, I see what you were suggesting: if you have `fromAddress(mAddr, seq)` then you *create* an underlying `Account` instance with the values set accordingly.. I like that!

Reply:
Exactly. it means that underlying base account is secretly not connected to anything. But I feel like that could be covered by a warning in the MuxedAccount.fromAddress docs.

Reply:
Unfortunately adapting the API to these changes (which I agree are better) introduces a circular dependency: Account now needs MuxedAccount to `createSubaccount()`, while MuxedAccount needs Account to `fromAddress`. I've merged them into a single file to avoid this, but I'm open to suggestions on how else to work around that.

Reply:
Eh. good enough for now. Let's clean it up later.